hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -488,7 +505,10 @@ private boolean doConnectToCandidateCluster(CandidateClusterContext context, boo
                 for (Member member : memberList) {
                     checkClientActive();
                     triedAddressesPerAttempt.add(member.getAddress());
-                    Connection connection = connect(member, o -> getOrConnectToMember((Member) o, switchingToNextCluster));
+                    connectionProcessListener.attemptingToConnectToAddress(translate(member.getAddress()));
```

Comment:
This is not the address we would connect if the public address information of the member is available.

Reply:
And, overall, I believe translating the address, which could end up calling blocking code, just because MC might add a listener is not good for the users who don't add such a listener. I believe, we need to either
- call the listener with non-translated address (I don't know if this will work for your use case)
- calling translate only if there are connection process listeners added

Reply:
added a check